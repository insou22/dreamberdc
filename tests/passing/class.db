class Player {
   const var health = 10!
}

const var player1 = new Player()!
const var player2 = new Player()! //Error: Can't have more than one 'Player' instance!