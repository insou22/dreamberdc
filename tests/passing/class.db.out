Program {
    items: [
        ClassDecl(
            ClassDecl {
                keyword: "class",
                name: Ident {
                    name: "Player",
                },
                body: [
                    Statement(
                        Assignment(
                            AssignmentStatement {
                                decl_type: ConstVar,
                                lhs: Ident(
                                    Ident {
                                        name: "health",
                                    },
                                ),
                                rhs: NumericLit(
                                    NumericLit {
                                        digits: "10",
                                    },
                                ),
                                termination: Importance(
                                    1,
                                ),
                            },
                        ),
                    ),
                ],
            },
        ),
        Statement(
            Assignment(
                AssignmentStatement {
                    decl_type: ConstVar,
                    lhs: Ident(
                        Ident {
                            name: "player1",
                        },
                    ),
                    rhs: New(
                        NewExpr {
                            class_name: Ident {
                                name: "Player",
                            },
                            params: [],
                        },
                    ),
                    termination: Importance(
                        1,
                    ),
                },
            ),
        ),
        Statement(
            Assignment(
                AssignmentStatement {
                    decl_type: ConstVar,
                    lhs: Ident(
                        Ident {
                            name: "player2",
                        },
                    ),
                    rhs: New(
                        NewExpr {
                            class_name: Ident {
                                name: "Player",
                            },
                            params: [],
                        },
                    ),
                    termination: Importance(
                        1,
                    ),
                },
            ),
        ),
    ],
}